### PR TITLE
Wip/hughsie/efiboot

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: contrib/ci/build_windows.sh [hunk 1]
```diff
@@ -42,6 +42,7 @@ meson setup .. \
     --bindir="bin" \
     -Dbuild=all \
     -Dman=false \
+    -Dplugin_uefi_capsule=disabled \
```
Comment (superm1) [inline contrib/ci/build_windows.sh:45]: shouldn't we be using `disable_auto_if()`?

File: contrib/snap/libefivar-fixpkgconfig/Makefile
Comment (superm1) [inline contrib/snap/libefivar-fixpkgconfig/Makefile:1]: I think the whole file can go if libefiboot is going.  As can the call to the part in the snapcraft yaml.

File: contrib/ci/build_windows.sh (hunk 1 above)
Comment (hughsie) [reply contrib/ci/build_windows.sh:45]: Something like?:

    -if get_option('plugin_uefi_capsule').allowed()
    +if get_option('plugin_uefi_capsule').disable_auto_if(host_machine.system() not in ['linux', 'freebsd']).allowed()

Comment (hughsie) [reply contrib/ci/build_windows.sh]: That seemed to work, thanks.

File: contrib/fwupd.spec.in [hunk 2]
```diff
@@ -100,7 +100,6 @@ BuildRequires: libmbim-devel
 %endif
 
 %if 0%{?have_uefi}
-BuildRequires: efivar-devel >= 33
```
Comment (superm1) [inline contrib/fwupd.spec.in:103]: something similar needs to be changed from `contrib/debian/rules`.

File: meson.build [hunk 3]
```diff
@@ -399,30 +399,28 @@ if host_machine.system() == 'freebsd'
...
-          EFI_MACHINE_TYPE_NAME = ''
+    EFI_MACHINE_TYPE_NAME = ''
   endif
   conf.set_quoted('EFI_MACHINE_TYPE_NAME', EFI_MACHINE_TYPE_NAME)
+endif
 
+if build_standalone and get_option('plugin_uefi_capsule').enabled()
   if get_option('efi_binary')
     efi_binary = dependency ('fwupd-efi', fallback: ['fwupd-efi', 'fwupd_efi_dep'])
   endif
```
Comment (superm1) [inline meson.build:423]: I think this block is missing the disable auto if stuff. how about doing this instead:
```
uefi_capsule = get_option('plugin_uefi_capsule').disable_auto_if(host_machine.system() not in ['linux', 'freebsd']).allowed()
if build_standalone and uefi_capsule
```

Then use `if uefi_capsule` in the plugins folder

Comment (superm1) [reply meson.build:423]: Or move this whole block into the plugins folder

File: contrib/fwupd.spec.in (hunk 2 above)
Comment (hughsie) [reply contrib/fwupd.spec.in:103]: I think I've done that.

File: meson.build (hunk 3 above)
Comment (hughsie) [reply meson.build:423]: > Or move this whole block into the plugins folder

That's probably the best thing to do; done -- lets see what CI thinks.

**Comment by hughsie** [comment]
@superm1 I've fixed up all the bugs and now fwupd produces exactly the same EFI keys as efiboot. I've even used this branch to reinstall my system firmware this morning :) Can you have a try on your system and see how it works for you? Thanks.

**Comment by superm1** [comment]
It worked to build exact same entries for me too, looks good thanks!

**Pull request merged**